src/Akeneo/Connectivity/Connection/back/Infrastructure/Apps/Controller/Public/ScopesUpdateAction.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Infrastructure\Apps\Controller\Public;
+
+use Akeneo\Connectivity\Connection\Application\Apps\Command\FlagAppContainingOutdatedScopesCommand;
+use Akeneo\Connectivity\Connection\Application\Apps\Command\FlagAppContainingOutdatedScopesHandler;
+use Akeneo\Connectivity\Connection\Domain\Apps\Persistence\FindOneConnectedAppByUserIdQueryInterface;
+use Akeneo\UserManagement\Component\Model\UserInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class ScopesUpdateAction
+{
+    public function __construct(
+        private TokenStorageInterface $tokenStorage,
+        private FindOneConnectedAppByUserIdQueryInterface $findOneConnectedAppByUserIdQuery,
+        private FlagAppContainingOutdatedScopesHandler $flagAppContainingOutdatedScopesHandler,
+    ) {
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        /** @var UserInterface|null $user */
+        $user = $this->tokenStorage->getToken()?->getUser();
+        if (null === $user) {
+            throw new AccessDeniedHttpException('Not an authenticated App');
+        }
+
+        $connectedApp = $this->findOneConnectedAppByUserIdQuery->execute($user->getId());
+        if (null === $connectedApp) {
+            throw new AccessDeniedHttpException('Not an authenticated App');
+        }
+
+        $this->flagAppContainingOutdatedScopesHandler->handle(new FlagAppContainingOutdatedScopesCommand(
+            $connectedApp,
+            $request->get('scopes', ''),
```

Comment:
```suggestion
            $request->query->get('scopes', ''),
```

Since symfony 5.4, it's deprecated to use the magic `$request->get()` method, it's better to use the specific InputBag.